File: Sources/KillerSudokuCore/Cell.swift
public struct Cell: Sendable, Codable {
    public var digit: Int?
    public var pencilMarks: Set<Int> = []
    /// True for a cell [[PuzzleGenerator]] pre-filled (ADR 0008) rather than one the player set —
    /// `Board.setDigit` refuses to change a given, matching standard Sudoku convention.
    public var isGiven: Bool = false

    public init(digit: Int? = nil, isGiven: Bool = false) {
        self.digit = digit
        self.isGiven = isGiven
    }
}

File: desktop-kotlin/src/main/kotlin/me/gileskayo/killersudoku/core/Cell.kt
package me.gileskayo.killersudoku.core

import kotlinx.serialization.Serializable

@Serializable
data class Cell(
    val digit: Int? = null,
    val pencilMarks: Set<Int> = emptySet(),
    /** True for a cell [PuzzleGenerator] pre-filled rather than one the player set --
     * [Board.setDigit] refuses to change a given, matching standard Sudoku convention. */
    val isGiven: Boolean = false,
)
